DiscordChatExporter.Core.Models/Guild.cs

```diff
@@ -26,13 +36,23 @@ public Guild(string id, string name, string? iconHash)
 
     public partial class Guild
     {
+        public static string GetUserColor(Guild guild, User user) => 
+                guild.Members.GetValueOrDefault(user.Id, null)?.Roles
+                ?.Select(r => guild.Roles
+                            .Where(role => r == role.Id)
+                            .FirstOrDefault()
+                )?.Where(r => r.Color != 0)?
+                .Aggregate<Role, Role?>(null, (a, b) => (a?.Position ?? 0) > b.Position? a : b)?
+                .ColorAsHex ?? "";
+        public static string GetUserNick(Guild guild, User user) => guild.Members.GetValueOrDefault(user.Id)?.Nick ?? user.Name;
+
         public static string GetIconUrl(string id, string? iconHash)
         {
             return !string.IsNullOrWhiteSpace(iconHash)
                 ? $"https://cdn.discordapp.com/icons/{id}/{iconHash}.png"
                 : "https://cdn.discordapp.com/embed/avatars/0.png";
         }
 
-        public static Guild DirectMessages { get; } = new Guild("@me", "Direct Messages", null);
+        public static Guild DirectMessages { get; } = new Guild("@me", "Direct Messages", new List<Role>(), null);
```

Comment:
Change `new List<Role>()` to `Array.Empty<Role>` after updating the constructor (re previous comment).